### ISSUE TITLE
Consider renaming `fit` method

### ISSUE THREAD

**Issue by jonathanng** [automl/auto-sklearn#851]
Because `fit` optimizes the machine learning models and builds an ensemble out of them, it cannot be used for basic sklearn functions such as `cross_val_predict`.

https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_predict.html

`cross_val_predict` would call `fit` for every cv split, not `refit`.

Any ideas on how to get around this?

**Comment by mfeurer** [comment]
> Because fit optimizes the machine learning models and builds an ensemble out of them, it cannot be used for basic sklearn functions such as cross_val_predict.

**Comment by mfeurer** [comment]
> Because fit optimizes the machine learning models and builds an ensemble out of them, it cannot be used for basic sklearn functions such as cross_val_predict.

I think it can be used with `cross_val_predict` to get a cross-validated estimate of the performance of Auto-sklearn (in the sense of doing a nested cross-validation). It will not work when using cross-validation inside of Auto-sklearn, too, because then `refit` is not executed. However, using `fit` for hyperparameter optimization is in line with scikit-learn, where [the `fit` method of GridSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV.fit) also triggers hyperparameter optimization.

> Any ideas on how to get around this?

Could you please describe what your goal is?

**Comment by jonathanng** [comment]
GridSearchCV gets around this by exposing a `best_estimator_` attribute. This can be used in `cross_val_predict`, and it will work as expected: hyperparameter tuning will not happen, the best model is just being fitted to new data.

From what I understand, `AutoSklearnClassifier` does not have a similar method. Is that correct?

**Comment by mfeurer** [comment]
That's correct, the closest thing is the method https://automl.github.io/auto-sklearn/master/api.html#autosklearn.classification.AutoSklearnClassifier.get_models_with_weights

It would be great to be compatible to that protocol, so if you want to create a PR that would be great. Another suggestion to get closer to the sklearn interface: add a refit argument to fit or init, which then triggers a refit at the end of fit (at the cost of running over the allocated time).
